src/ansys/fluent/core/workflow_new.py

```diff
@@ -0,0 +1,1024 @@
+simulation workflows, with automatic dependency management and validation.
+"""
+
+from __future__ import annotations
+
+from collections import OrderedDict
+import re
+
+from ansys.fluent.core import config
+from ansys.fluent.core.services.datamodel_se import PyMenu, PyMenuGeneric
+from ansys.fluent.core.utils import load_module
+from ansys.fluent.core.utils.fluent_version import FluentVersion
+
+
+def _convert_task_list_to_display_names(
+    workflow_root: PyMenu, task_list: list[str]
+) -> list[str]:
+    """Convert a list of task IDs to their corresponding display names.
+
+    Parameters
+    ----------
+    workflow_root : PyMenu
+        The root workflow datamodel object that provides service access.
+    task_list : list[str]
+        List of internal task identifiers (e.g., ["TaskObject1", "TaskObject2"]).
+
+    Returns
+    -------
+    list[str]
+        List of display names corresponding to the task IDs
+        (e.g., ["Import Geometry", "Add Local Sizing"]).
+    """
+    _display_names = []
+    for _task_name in task_list:
+        name_obj = PyMenu(
+            service=workflow_root.service,
+            rules=workflow_root.rules,
+            path=[("task_object", _task_name), ("_name_", "")],
+        )
+        _display_names.append(name_obj.get_remote_state())
+    return _display_names
+
+
+def _get_child_task_by_task_id(workflow_root, task_id):
+    """Get a child task's display name by its internal task ID.
+
+    Parameters
+    ----------
+    workflow_root : PyMenu
+        The root workflow datamodel object.
+    task_id : str
+        Internal identifier for the task (e.g., "TaskObject1").
+
+    Returns
+    -------
+    str
+        The display name of the task (e.g., "Import Geometry").
+    """
+    return PyMenu(
+        service=workflow_root.service,
+        rules=workflow_root.rules,
+        path=[("task_object", task_id), ("_name_", "")],
+    ).get_remote_state()
+
+
+def command_name_to_task_name(workflow_root, command_name: str) -> str:
+    """Convert a command name to its corresponding task display name.
+
+    This function maps internal command names (used by the Fluent core) to
+    user-facing task names.
+
+    Parameters
+    ----------
+    workflow_root : PyMenu
+        The root workflow datamodel object.
+    command_name : str
+        Internal command name (e.g., "ImportGeometry").
+
+    Returns
+    -------
+    str
+        User-facing task name (e.g., "import_geometry").
+
+    Notes
+    -----
+    This is a workaround for Fluent 26R1. It attempts to load version-specific
+    datamodel files to perform the conversion, falling back to a generic approach if unavailable.
+    """
+    # TODO: This is a fix only for 26R1 as the server lacks the mechanism to return mapped values
+    #  for '<Task Object>.get_next_possible_tasks()'.
+    try:
+        module = load_module(
+            "meshing_261", config.codegen_outdir / "datamodel_261" / "meshing.py"
```

Comment:
I have two suggestions here.
1. Pass in all dependencies from above; those are the `meshing_workflow` and `meshing` data model instances.
2. Access the `meshing` data model instance in the usual way rather than breaking encapsulation by mentioning `codegen_outdir` etc.

Reply:
I wanted to make the interface with minimal change, but yes the design was flawed. I'll update according to your suggestions.